src/middlewared/middlewared/plugins/sysdataset.py

```diff
@@ -244,129 +259,136 @@ async def destination_pool_error(self, new_pool):
             if job.error:
                 raise CallError(job.error)
             return
 
         # If we dont have a pool configured in the database try to find the first data pool
         # to put it on.
         if not config['pool_set']:
-            if pool := await self._query_pool_for_system_dataset(exclude_pool):
+            if pool := self._query_pool_for_system_dataset(exclude_pool):
                 self.logger.debug('Sysdataset pool was not set, moving it to first available pool %r', pool['name'])
-                job = await self.middleware.call('systemdataset.update', {'pool': pool['name']})
-                await job.wait()
+                job = self.middleware.call_sync('systemdataset.update', {'pool': pool['name']})
+                job.wait_sync()
                 if job.error:
                     raise CallError(job.error)
                 return
 
+        mntinfo = self.middleware.call_sync('filesystem.mount_info')
         if config['pool'] != boot_pool:
-            dataset = await self.middleware.call('pool.dataset.query', [['name', '=', config['pool']]],
-                                                 {'extra': {'retrieve_children': False}})
-            if not dataset or (dataset[0]['locked'] and dataset[0]['key_format']['value'] != 'PASSPHRASE'):
+            if not any(filter_list(mntinfo, [['mount_source', '=', config['pool']]])):
                 # Pool is not mounted (e.g. HA node B), temporary set up system dataset on the boot pool
                 self.logger.debug(
                     'Root dataset for pool %r is not available, temporarily setting up system dataset on boot pool',
                     config['pool'],
                 )
                 self.force_pool = boot_pool
-                config = await self.config()
+                config = self.middleware.call_sync('systemdataset.config')
 
         mounted_pool = mounted = None
-        for p in psutil.disk_partitions():
-            if p.mountpoint == SYSDATASET_PATH:
-                mounted_pool = p.device.split('/')[0]
-        if mounted_pool and mounted_pool != config['pool']:
+
+        sysds_mntinfo = filter_list(mntinfo, [['mountpoint', '=', '/var/db/system']])
+        if sysds_mntinfo:
+            mounted_pool = sysds_mntinfo[0]['mount_source']
+
+        if mounted_pool and mounted_pool.split('/')[0] != config['pool']:
             self.logger.debug('Abandoning dataset on %r in favor of %r', mounted_pool, config['pool'])
-            async with self._release_system_dataset():
-                await self.__umount(mounted_pool, config['uuid'])
-                await self.__setup_datasets(config['pool'], config['uuid'])
-                mounted = await self.__mount(config['pool'], config['uuid'])
+            with self._release_system_dataset():
+                self.__umount(mounted_pool, config['uuid'])
+                self.middleware.call_sync('systemdataset.setup_datasets', config['pool'], config['uuid'])
```

Comment:
Missing underscore in called method name.
Does this pass tests? If yes, we need the one that fails here.

Reply:
It's better not to have any underscore at all. `@private` and that's it.